### PR TITLE
[Test Proxy] Normalize paths in test IDs

### PR DESCRIPTION
**Pull request by mccoyp**
# Description

Context: test IDs, which are used by the proxy to determine recording locations, are currently determined based on the value of the `pytest`-controlled `PYTEST_CURRENT_TEST` environment variable. This variable's value isn't consistent across environments though -- the path to the test file is sometimes relative to the repository root, and sometimes not. See https://github.com/Azure/azure-sdk-for-python/issues/22387#issuecomment-1011680482 for more details.

To adjust to this variation in the environment variable, this PR adds logic to determine the repo-root relative path to the current test. This ensures that the test ID is always consistent.

**Note:** AzureRecordedTestCase's `create_random_name` method also relies on the value of `PYTEST_CURRENT_TEST`. The inconsistency of this variable could result in different pseudo-random resource names, which could cause test failures in playback. Fixing this is tracked by https://github.com/Azure/azure-sdk-for-python/issues/22512.

@scbedd for notifs

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../CONTRIBUTING.md##building-and-testing)
- _N/A_ Pull request includes test coverage for the included changes.
  - An already-migrated test suite is used to verify these changes

### CODE REVIEW

File: tools/azure-sdk-tools/devtools_testutils/proxy_testcase.py [hunk 1]
```diff
@@ -57,11 +57,21 @@ def get_recording_id():
 def get_test_id():
     # type: () -> str
     # pytest sets the current running test in an environment variable
+    # the path to the test can depend on the environment, so we can't assume this is the path from the repo root
     setting_value = os.getenv("PYTEST_CURRENT_TEST")
-
     path_to_test = os.path.normpath(setting_value.split(" ")[0])
-    path_components = path_to_test.split(os.sep)
-
+    full_path_to_test = os.path.abspath(path_to_test)
```
Comment (scbedd) [inline tools/azure-sdk-tools/devtools_testutils/proxy_testcase.py:63]: <snipped from problem description so I can compare to code base>

`sdk/search/azure-search-documents/tests/test_search_indexer_client_live.py::TestSearchIndexersClient::test_delete_indexer_if_unchanged (call)'`

When debugging the same test via VS Code, PYTEST_CURRENT_TEST is
`tests/test_search_indexer_client_live.py::TestSearchIndexersClient::test_delete_indexer_if_unchanged (call)`.

File: tools/azure-sdk-tools/devtools_testutils/proxy_testcase.py [hunk 2]
```diff
@@ -57,11 +57,21 @@ def get_recording_id():
...
+
+    # walk up to the repo root by looking for "sdk" directory or root of file system
+    path_components = []
+    head, tail = os.path.split(full_path_to_test)
+    while tail != "sdk" and tail != "":
```
Comment (scbedd) [inline tools/azure-sdk-tools/devtools_testutils/proxy_testcase.py:68]: I don't _think_ you have any other stop cases here. Famous last words 👍

**Review by scbedd — APPROVED** [review]

File: tools/azure-sdk-tools/devtools_testutils/proxy_testcase.py (hunk 2 above)
Comment (mccoyp) [reply tools/azure-sdk-tools/devtools_testutils/proxy_testcase.py:68]: Writing a `while` loop always terrifies me, but I'm also _pretty_ sure this should be good. Knocking furiously on wood

**Review by tjprescott — APPROVED** [review]
If it unblocks me, I'm happy with it :)

**Pull request merged**